File: mp-b/admin/api/outward.ts
/**
 * 外出登记相关接口
 * 对应后端 OutwardController（/outward/...）
 */
import { get, post } from '../utils/request'
import { PageResult, mockPage } from './elder'

const PAGE_NUM = 1
const PAGE_SIZE = 200

export interface OutwardItem {
	id: number
	elder_name: string
	chaperone_name: string
	chaperone_phone: string
	chaperone_type: string
	outward_date: string
	plan_return_date: string
	real_return_date: string
}

export interface OutwardForm {
	id?: number
	elder_id: number
	chaperone_name: string
	chaperone_phone: string
	chaperone_type: string
	outward_date: string
	plan_return_date: string
}

// ====== MOCK 数据（后台未完成，测试前端用；后台完成后放开各函数内注释并删除本段）======
const mockOutwardList: OutwardItem[] = [
	{ id: 1, elder_name: '张建国', chaperone_name: '张小明', chaperone_phone: '13900000001', chaperone_type: '家属', outward_date: '2026-08-20 09:00', plan_return_date: '2026-08-20 18:00', real_return_date: '' },
	{ id: 2, elder_name: '王秀兰', chaperone_name: '王大力', chaperone_phone: '13900000003', chaperone_type: '家属', outward_date: '2026-08-21 10:00', plan_return_date: '2026-08-22 17:00', real_return_date: '2026-08-21 16:00' }
]

/** 外出列表（全量） */
export function pageOutwardByKey(elderName?: string, chaperoneType?: string): Promise<PageResult<OutwardItem>> {
	// return get<PageResult<OutwardItem>>('/outward/pageOutwardByKey', {
	// 	page_num: PAGE_NUM,
	// 	page_size: PAGE_SIZE,
	// 	elder_name: elderName || undefined,
	// 	chaperone_type: chaperoneType || undefined
	// })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	let list = mockOutwardList
	if (elderName) list = list.filter((o) => o.elder_name.indexOf(elderName) >= 0)
	if (chaperoneType) list = list.filter((o) => o.chaperone_type === chaperoneType)
	return Promise.resolve(mockPage(list))
}

export function getOutwardById(id: number): Promise<OutwardItem> {
	// return get<OutwardItem>('/outward/getOutwardById', { id })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve(mockOutwardList.find((o) => o.id === id) || mockOutwardList[0])
}

export function addOutward(form: OutwardForm): Promise<void> {
	// return post<void>('/outward/addOutward', form)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

export function editOutward(form: OutwardForm): Promise<void> {
	// return post<void>('/outward/editOutward', form)
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

export function delayReturn(id: number, planReturnDate: string): Promise<void> {
	// return post<void>('/outward/delayReturn', { id, plan_return_date: planReturnDate })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

export function recordReturn(id: number, realReturnDate: string): Promise<void> {
	// return post<void>('/outward/recordReturn', { id, real_return_date: realReturnDate })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

export function deleteOutward(id: number): Promise<void> {
	// return post<void>('/outward/deleteOutward', { id })
	// ====== MOCK（后台完成并放开上方注释后，可删除此段）======
	return Promise.resolve()
}

File: backend/internal/model/dto/outward.go
package dto

import "time"

// ============ OutwardController 请求 ============

// PageOutwardByKeyReq 分页查询外出登记请求
// @request
type PageOutwardByKeyReq struct {
	PageNum       *int       `json:"page_num" valid:"required"`  // 页码
	PageSize      *int       `json:"page_size" valid:"required"` // 条数
	ElderName     *string    `json:"elder_name"`                 // 老人姓名
	ChaperoneType *string    `json:"chaperone_type"`             // 陪同人类型
	StartTime     *time.Time `json:"start_time"`                 // 开始时间
	EndTime       *time.Time `json:"end_time"`                   // 结束时间
}

// AddOutwardReq 新增外出登记请求
// @request
type AddOutwardReq struct {
	ElderID        *int64     `json:"elder_id" valid:"required"`         // 老人编号
	ChaperoneName  *string    `json:"chaperone_name" valid:"required"`   // 陪同人姓名
	ChaperonePhone *string    `json:"chaperone_phone" valid:"required"`  // 陪同人电话
	ChaperoneType  *string    `json:"chaperone_type" valid:"required"`   // 陪同人类型
	OutwardDate    *time.Time `json:"outward_date" valid:"required"`     // 外出时间
	PlanReturnDate *time.Time `json:"plan_return_date" valid:"required"` // 计划返回时间
}

// DelayReturnReq 延期返回请求
// @request
type DelayReturnReq struct {
	ID             *int64     `json:"id" valid:"required"`               // id
	PlanReturnDate *time.Time `json:"plan_return_date" valid:"required"` // 计划返回时间
}

// RecordReturnReq 登记返回请求
// @request
type RecordReturnReq struct {
	ID             *int64     `json:"id" valid:"required"`               // id
	RealReturnDate *time.Time `json:"real_return_date" valid:"required"` // 实际返回时间
}

// EditOutwardReq 编辑外出登记请求
// @request
type EditOutwardReq struct {
	ID             *int64     `json:"id"`                                // id
	ElderID        *int64     `json:"elder_id" valid:"required"`         // 老人编号
	ChaperoneName  *string    `json:"chaperone_name" valid:"required"`   // 陪同人姓名
	ChaperonePhone *string    `json:"chaperone_phone" valid:"required"`  // 陪同人电话
	ChaperoneType  *string    `json:"chaperone_type" valid:"required"`   // 陪同人类型
	OutwardDate    *time.Time `json:"outward_date" valid:"required"`     // 外出时间
	PlanReturnDate *time.Time `json:"plan_return_date" valid:"required"` // 计划返回时间
}

// ============ OutwardController 响应 ============

// PageOutwardByKeyResp 分页查询外出登记响应
// @response
type PageOutwardByKeyResp struct {
	ID             int64     `json:"id"`               // 外出登记编号
	ElderName      string    `json:"elder_name"`       // 老人姓名
	ChaperoneName  string    `json:"chaperone_name"`   // 陪同人姓名
	ChaperonePhone string    `json:"chaperone_phone"`  // 陪同人电话
	ChaperoneType  string    `json:"chaperone_type"`   // 陪同人类型
	OutwardDate    time.Time `json:"outward_date"`     // 外出时间
	PlanReturnDate time.Time `json:"plan_return_date"` // 计划返回时间
	RealReturnDate time.Time `json:"real_return_date"` // 实际返回时间
}

// GetOutwardByIDResp 根据编号查询外出登记响应
// @response
type GetOutwardByIDResp struct {
	ID             int64     `json:"id"`               // id
	ElderName      string    `json:"elder_name"`       // 老人姓名
	ChaperoneName  string    `json:"chaperone_name"`   // 陪同人姓名
	ChaperonePhone string    `json:"chaperone_phone"`  // 陪同人电话
	ChaperoneType  string    `json:"chaperone_type"`   // 陪同人类型
	OutwardDate    time.Time `json:"outward_date"`     // 外出时间
	PlanReturnDate time.Time `json:"plan_return_date"` // 计划返回时间
}
